modules/programs/btop.nix

```diff
@@ -0,0 +1,54 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.btop;
+
+  formatLine = n: v:
+    let
+      formatValue = v:
+        if builtins.isBool v then
+          (if v then "True" else "False")
+        else if builtins.isString v then
+          ''"${v}"''
+        else
+          toString v;
+    in "${n} = ${formatValue v}";
+in {
+  meta.maintainers = [ hm.maintainers.GaetanLepage ];
+
+  options.programs.btop = {
+    enable = mkEnableOption "btop";
+
+    settings = mkOption {
+      type = with types; attrsOf (oneOf [ bool float int str ]);
+      default = { };
+      description = ''
+        Options to add to <filename>btop.conf</filename> file.
+        <link xlink:href="https://github.com/aristocratos/btop#configurability"/>
+        for options.
+      '';
+      example = {
+        color_theme = "Default";
+        theme_background = false;
+      };
+    };
+
+    extraConfig = mkOption {
+      type = types.lines;
+      default = "";
+      description = ''
+        Extra lines added to <filename>btop.conf</filename> file.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ pkgs.btop ];
+
+    xdg.configFile."btop/btop.conf".text = concatStringsSep "\n" ([ ]
```

Comment:
Most programs have a `mkIf (cfg.settings != {})`, so the file doesn't get created if there are no settings. May that be desirable here as well?

Reply:
Indeed, I just added it !